### PR TITLE
Fix TsFileResource endTime error when insert empty tablet and flush

### PR DESCRIPTION
**Pull request by HTHou**
## Description

When data is written using the insertTablet() interface and both values are null, exactly the data written is flushed. An empty TsFile file will be generated, the space cannot be deleted, and the generated resource file will be incorrect, and the endTime of the device will be Long.MIN_VALUE. This in turn affects query and merge mechanisms.

Based on this, this PR has the following two optimizations:
1. In the flush process, it is added that if the resource file is empty, the generated TsFile and.resource file are deleted
2. Enhance the judgment logic when there is no valid data in MemTable. If there is no valid data, delete the related empty Device data from the resource.

## How to test
Execute the following code.

```java
  public static void main(String[] args)
      throws IoTDBConnectionException, StatementExecutionException {
    session =
        new Session.Builder()
            .host(LOCAL_HOST)
            .port(6667)
            .username("root")
            .password("root")
            .version(Version.V_1_0)
            .build();
    session.open(false);
    insertTablet();
    session.executeNonQueryStatement("flush");
    session.close();
  }


  private static void insertTablet() throws IoTDBConnectionException, StatementExecutionException {
    // The schema of measurements of one device
    // only measurementId and data type in MeasurementSchema take effects in Tablet
    List<MeasurementSchema> schemaList = new ArrayList<>();
    schemaList.add(new MeasurementSchema("s1", TSDataType.INT64));
    schemaList.add(new MeasurementSchema("s2", TSDataType.INT64));
    schemaList.add(new MeasurementSchema("s3", TSDataType.INT64));

    Tablet tablet = new Tablet(ROOT_SG1_D1, schemaList, 100);

    // Method 1 to add tablet data
    long timestamp = System.currentTimeMillis();

    for (long row = 0; row < 100; row++) {
      int rowIndex = tablet.rowSize++;
      tablet.addTimestamp(rowIndex, timestamp);
      for (int s = 0; s < 3; s++) {
        tablet.addValue(schemaList.get(s).getMeasurementId(), rowIndex, null);
      }
      if (tablet.rowSize == tablet.getMaxRowNumber()) {
        session.insertTablet(tablet, true);
        tablet.reset();
      }
      timestamp++;
    }

    if (tablet.rowSize != 0) {
      session.insertTablet(tablet);
      tablet.reset();
    }
  }

```

Then print the TsFileResource file, you will see the endTime is Long.MIN_VALUE.

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/DataRegion.java [hunk 1]
```diff
@@ -2053,7 +2053,7 @@ private void closeUnsealedTsFileProcessorCallBack(TsFileProcessor tsFileProcesso
     closeQueryLock.writeLock().lock();
     try {
       tsFileProcessor.close();
-      if (tsFileProcessor.isEmpty()) {
+      if (tsFileProcessor.isEmpty() || tsFileProcessor.getTsFileResource().isEmpty()) {
```
Comment (wangchao316) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/DataRegion.java:2056]: hi , tsfile exsit and resource does not exsit,  will delete tsfile?

Comment (HTHou) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/DataRegion.java:2056]: In this case, tsfile contains no data and resource file exists. 
I'll update the PR description later.

Comment (wangchao316) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/DataRegion.java:2056]: ok

**Review by wangchao316 — APPROVED** [review]
LGTM

**Review by xingtanzjr — APPROVED** [review]

**Review by caozj1011 — APPROVED** [review]

**Pull request merged**